controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Clusters.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.resource;
+
+import com.pinterest.rocksplicator.controller.bean.TaskBean;
+
+import java.util.List;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+@Path("/v1/clusters/{clusterName : [a-zA-Z0-9\\-_]+}")
+public class Clusters {
+
+  /**
+   * Retrieves all tasks belong to a given cluster by cluster name.
+   *
+   * @param clusterName name of the cluster
+   * @return            a list of {@link TaskBean}s belong to clusterName
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<TaskBean> getAllTasks(@PathParam("clusterName") String clusterName) {
+    throw new UnsupportedOperationException("method not implemented.");
+  }
+
+  /**
+   * Aborts the current running tasks on a given cluster.
+   *
+   * @param clusterName name of the cluster
+   * @param retry       whether to retry abort or not if the first attempt failed
+   * @return            a list of {@link TaskBean}s aborted
+   */
+  @POST
+  @Path("/abortTask")
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<TaskBean> abortTask(@PathParam("clusterName") String clusterName,
+                                  @QueryParam("retry") boolean retry) {
+    throw new UnsupportedOperationException("method not implemented.");
+  }
+
+  /**
+   * Initializes the shard config and controller config for a given cluster.
+   *
+   * @param clusterName name of the cluster
+   */
+  @POST
+  @Path("/initialize")
+  public void initialize(@PathParam("clusterName") String clusterName) {
+    throw new UnsupportedOperationException("method not implemented.");
+  }
+
+  /**
+   * Expands or shrinks the cluster to the given size.
+   *
+   * @param clusterName name of the cluster
+   * @param size        target cluster size
+   */
+  @POST
+  @Path("/resize")
+  public void resize(@PathParam("clusterName") String clusterName,
+                     @QueryParam("size") int size) {
+    throw new UnsupportedOperationException("method not implemented.");
+  }
+
+
+  /**
+   * Recovers the config of a cluster to the last snapshot.
+   *
+   * @param clusterName name of the cluster
+   */
+  @POST
+  @Path("/recoverConfig")
```

Comment:
in case something goes wrong, we need a way to rollback to last correct shard mapping. i feel this api will be useful if some tasks enter bad state and we want to rollback to last good config